### DISCUSSION TITLE
Labwc and systemd user services

### DISCUSSION THREAD

**Discussion by oct2pus** [labwc/labwc#2047] [Show and tell]
i like managing services from systemd so that i can troubleshoot any errors with running services from journalctl.

I just created a [labwc-session.target](https://git.jade.moe/oct2pus/labwc-services/raw/branch/main/labwc-session.target) slightly modified from an [arch wiki segment on sway & systemd](https://wiki.archlinux.org/title/Sway#Manage_Sway-specific_daemons_with_systemd) and modified [autostart](https://git.jade.moe/oct2pus/dotfiles/src/branch/main/labwc/autostart) to be start the link target. 
you can look at a (slightly outdated out the time of writing) list of [services](https://git.jade.moe/oct2pus/labwc-services) i'm running as an example of what a service should look like.

**Comment by stefonarch** [comment]
I used sometimes in my start script (LXQt) just
` exec systemd-cat --identifier=labwc labwc -d  -S lxqt-session`

**Comment by epyonavenger** [comment]
I did something similar for quite a while before LXQt caught up.

I found it worked pretty well other than you had to get good at figuring out what services needed what variables pushed out with `dbus-update-activation-environment` before other services so that it all played nice.

It'd be nice if one of the DEs started leveraging systemd user services a little more instead of essentially re-implementing the wheel with their own session/process watchers/restarters.

**Reply by fungilife** [reply]
that would mean also doing something similar for sysV, runit, OpenRC, s6, dinit, and have the pkg be configured for a specific service-supervisor/manager/init 

On the other hand it appears to be working fine without a logind and dbus as a window-manager
DEs are a whole different game irrelevant to the wm laywer.  DE is something you build on top of a wm, which is what labwc is.

**Reply by epyonavenger** [reply]
I'm not sure how that's relevant to the OP, or my comment. OP was looking at using systemd to do things, stefonarch and I were commenting on how one could do that, and I then followed up with how it would be nice if DEs just used systemd properly instead of reinventing the wheel with their own service manager.

None of us here brought up other/legacy service managers. For people who want to go the old-school/manual route, I imagine they have the skill to do so, or else they wouldn't be sticking to legacy service managers.

Either way, we're essentially building or own DEs on top of LabWC, which is why the auto-start/service-manager aspect is desirable. Starting LabWC with systemd (as in stefonarch's example), then hanging the other services off its target (as in OP's comment) with appropriate dependency chains in the services (and restart logic in the case of failures) is fantastic for a lightweight system.

In any case, lest I stray too far from _your_ comment with my own drabbles, the conversation is around using systemd. So I'm not sure discussion about any other legacy service managers is interesting here.
